iceoryx_hoofs/test/moduletests/test_vocabulary_semantic_string.cpp

```diff
@@ -0,0 +1,367 @@
+            {
+                auto sut = SutType::create(string<SutType::capacity()>(TruncateToCapacity, value.c_str()));
+                ASSERT_THAT(sut.has_error(), Eq(false));
+
+                EXPECT_THAT(sut->insert(insert_position,
+                                        string<SutType::capacity()>(TruncateToCapacity, add_value.c_str()),
+                                        add_value.size())
+                                .has_error(),
+                            Eq(true));
+
+
+                EXPECT_THAT(sut->size(), Eq(value.size()));
+                EXPECT_THAT(sut->capacity(), Eq(TestValues<SutType>::CAPACITY));
+
+                EXPECT_THAT(sut->as_string().c_str(), StrEq(value));
+            }
+        }
+    }
+}
+
+TYPED_TEST(SemanticString_test, EqualityOperatorWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "97889932-ac3b-4155-9958-34c843d2d323");
+
+    EXPECT_TRUE(this->greater_value == this->greater_value);
+    EXPECT_FALSE(this->greater_value == this->smaller_value);
+
+    EXPECT_TRUE(this->greater_value == this->greater_value_str);
+    EXPECT_FALSE(this->greater_value == this->smaller_value_str);
+}
+
+TYPED_TEST(SemanticString_test, InequalityOperatorWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "32903b0b-3594-4c00-9869-d18e1dfc773f");
+
+    EXPECT_FALSE(this->greater_value != this->greater_value);
+    EXPECT_TRUE(this->greater_value != this->smaller_value);
+
+    EXPECT_FALSE(this->greater_value != this->greater_value_str);
+    EXPECT_TRUE(this->greater_value != this->smaller_value_str);
+}
+
+TYPED_TEST(SemanticString_test, LessThanOrEqualOperatorWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "53f5b765-b462-4cc1-bab7-9b937fbbcecf");
+
+    EXPECT_TRUE(this->greater_value <= this->greater_value);
+    EXPECT_TRUE(this->smaller_value <= this->greater_value);
+    EXPECT_FALSE(this->greater_value <= this->smaller_value);
+}
+
+TYPED_TEST(SemanticString_test, LessThanOperatorWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "cea977a4-ccb3-42a6-9d13-e09dce24c273");
+
+    EXPECT_FALSE(this->greater_value < this->greater_value);
+    EXPECT_TRUE(this->smaller_value < this->greater_value);
+    EXPECT_FALSE(this->greater_value < this->smaller_value);
+}
+
+TYPED_TEST(SemanticString_test, GreaterThanOrEqualOperatorWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5d731b17-f787-46fc-b64d-3d86c9102008");
+
+    EXPECT_TRUE(this->greater_value >= this->greater_value);
+    EXPECT_FALSE(this->smaller_value >= this->greater_value);
+    EXPECT_TRUE(this->greater_value >= this->smaller_value);
+}
+
+TYPED_TEST(SemanticString_test, GreaterThanOperatorWorks)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8c046cff-fb69-43b4-9a45-e86f17c874db");
+
+    EXPECT_FALSE(this->greater_value > this->greater_value);
+    EXPECT_FALSE(this->smaller_value > this->greater_value);
+    EXPECT_TRUE(this->greater_value > this->smaller_value);
+}
+} // namespace
```

Comment:
Should methods like `size`, `capacity` and `as_string` also be explicitly tested? I mean not all corner cases, just to check whether the call is correctly forwarded to the underlying `iox::string`.